presto-main/src/main/java/com/facebook/presto/operator/window/AggregateWindowFunction.java

```diff
@@ -80,7 +79,7 @@ else if ((frameStart == currentStart) && (frameEnd >= currentEnd)) {
     private void accumulate(int start, int end)
     {
         if (function.getParameterTypes().size() == 0) {
-            accumulator.addInput(new Page(end - start + 1));
+            accumulator.addInput(end - start + 1);
```

Comment:
So I think the optimization in this commit is entirely about eliminating this call to `new Page()`.  Does this change effect the performance of real queries?  If not, I don't think it is worth the extra complexity.